components/script/dom/textencoder.rs

```diff
@@ -68,4 +72,55 @@ impl TextEncoderMethods for TextEncoder {
         create_buffer_source(cx, encoded, js_object.handle_mut())
             .expect("Converting input to uint8 array should never fail")
     }
+
+    // https://encoding.spec.whatwg.org/#dom-textencoder-encodeinto
+    #[allow(unsafe_code)]
+    fn EncodeInto(
+        &self,
+        source: USVString,
+        mut destination: CustomAutoRooterGuard<typedarray::Uint8Array>,
+    ) -> TextEncoderEncodeIntoResult {
+        let available = destination.len();
+
+        // Bail out if the destination has no space available.
+        if available == 0 {
+            return TextEncoderEncodeIntoResult {
+                read: Some(0),
+                written: Some(0),
+            };
+        }
+
+        // Steps 1, 2
```

Comment:
Please put a description of what's at these steps, because the numbers change pretty frequently in the HTML specification.

Reply:
done!